os/fs/smartfs/smartfs_utils.c

```diff
@@ -1217,11 +1243,19 @@ int smartfs_createentry(struct smartfs_mountpt_s *fs, uint16_t parentdirsector,
 	/* Create the directory entry to be written in the parent's sector */
 
 #ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
-	smartfs_wrle16(&entry->firstsector, nextsector);
-	smartfs_wrle16(&entry->utc, time(NULL));
+	smartfs_wrle32(&entry->firstsector, nextsector);
+#ifdef CONFIG_UTC
+	smartfs_wrle32(&entry->utc, time(NULL));
+#else
+	smartfs_wrle32(&entry->utc, fs->entry_seq++);
+#endif
 #else
 	entry->firstsector = nextsector;
+#ifdef CONFIG_UTC
 	entry->utc = time(NULL);
+#else
+	entry->utc = fs->entry_seq++;
```

Comment:
Suppose smartfs is unmounted and remounted (device resets after power off).
fs->entry_seq is reset to 0.

Smartfs partition already has active entries with sequence numbers say 45, 57, 89

Now, these entries are to be replaced (due to file operations or other requests).
The new entries will have entry->utc = 0, 1, 2, 3....... which are smaller than old sequence numbers.

How will we recover base don this? We expect a higher sequence number to be the latest entry.

Reply:
It will updated during fs_recover.